File: previous/283.move-zeros/283.move-zeroes.js
/*
 * @lc app=leetcode id=283 lang=javascript
 *
 * [283] Move Zeroes
 * Time: O(n)
 * Space: O(1), in-place
 * Note: 先覆蓋, 最後再將 0 補回去
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let j = 0;
    for(let i=0; i<nums.length; i++){
        if(nums[i] != 0){
            nums[j] = nums[i];
            j++;
        }
    }

    for(let o=j; o<nums.length; o++){
        nums[o] = 0;
    }

        
    return nums;
};
// @lc code=end

File: 7-day-sprint/day1/283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        swap_pos = 0
        for i in range(0, len(nums)):
            if nums[i] != 0:
                nums[swap_pos], nums[i] = nums[i], nums[swap_pos]
                swap_pos += 1
        
# @lc code=end

# case
# [0,1,0]
# [0,0,1]
# [1,1,0]
# [0]
# [1]

